querydsl-sql-codegen/src/main/java/com/querydsl/sql/codegen/ExtendedBeanSerializer.java

```diff
@@ -112,4 +112,9 @@ protected void bodyEnd(EntityType model, CodeWriter writer) throws IOException {
 
     }
 
+    @Override
+    // In this class toString is added within bodyEnd, override with empty to ensure the base class does not add another
+    // toString
+    protected void addToString(EntityType model, CodeWriter writer) {
```

Comment:
This seems a bit hacky to me, can't we move the code from the `bodyEnd()` to this overridden `addToString`? Is the implementation different from the default anyway? The toString in `bodyEnd()` doesn't really seem that sophisticated. Rather more a debugging leftover...

Reply:
I tried doing this initially, but unfortunately `toString` is being built along with collecting properties / pks within `bodyEnd`. Extracting this would either result duplication of this logic or keeping the state between the method calls. Agree this solution is hacky but best I could come up without rewriting it too much and preserving the format at the same time.

The implementation is different, yes, `ExtendedBeanSerializer` only includes the primary keys into `toString`. And I cannot move this up to `BeanSerializer` either as it should not be aware of PKs.

toString example using the unit test:
```
BeanSerializer: id = 1, name = X, sub_id = ##
ExtendedBeanSerializer: DomainClass#1;##
```